core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-store/dot-pages.store.ts

```diff
@@ -78,10 +356,99 @@ export class DotPageStore extends ComponentStore<DotPagesState> {
+                        const wfActionEvent: DotCMSWorkflowActionEvent = {
+                            workflow: action,
+                            callback: 'ngWorkflowEventCallback',
+                            inode: item.inode,
+                            selectedInodes: null
+                        };
+                        this.dotWorkflowEventHandlerService.open(wfActionEvent);
+                    } else {
+                        this.dotWorkflowActionsFireService
+                            .fireTo(item.inode, action.id)
+                            .subscribe(() => {
+                                this.dotEventsService.notify<DotGlobalMessage>(
+                                    'dot-global-message',
+                                    {
+                                        value: this.dotMessageService.get('Workflow-executed')
+                                    }
+                                );
+                            });
+                    }
+                }
+            });
+        });
+
+        // Adding Add To Bundle action
+        if (isEnterprise) {
+            selectItems.push({
+                label: this.dotMessageService.get('contenttypes.content.add_to_bundle'),
+                command: () => this.showAddToBundle(item.identifier)
+            });
+        }
+
+        // Adding Push Publish action
+        if (isEnterprise && environments) {
+            selectItems.push({
+                label: this.dotMessageService.get('contenttypes.content.push_publish'),
+                command: (item) =>
+                    this.dotPushPublishDialogService.open({
+                        assetIdentifier: item.identifier,
+                        title: this.dotMessageService.get('contenttypes.content.push_publish')
+                    })
+            });
+        }
+
+        return selectItems;
+    }
+
     constructor(
         private dotCurrentUser: DotCurrentUserService,
+        private dotCurrentUserService: DotCurrentUserService,
+        private dotRouterService: DotRouterService,
         private httpErrorManagerService: DotHttpErrorManagerService,
-        private dotESContentService: DotESContentService
+        private dotESContentService: DotESContentService,
+        private dotFormatDateService: DotFormatDateService,
+        private dotMessageService: DotMessageService,
+        private dotLanguagesService: DotLanguagesService,
+        private dotPushPublishDialogService: DotPushPublishDialogService,
+        private dotWorkflowsActionsService: DotWorkflowsActionsService,
+        private dotWorkflowEventHandlerService: DotWorkflowEventHandlerService,
+        private dotWorkflowActionsFireService: DotWorkflowActionsFireService,
+        private dotLicenseService: DotLicenseService,
+        private dotEventsService: DotEventsService,
+        private pushPublishService: PushPublishService,
+        private siteService: SiteService
```

Comment:
This is crazy.... I know we need to put this out, but is there any way to split this store maybe?

Reply:
IMO, having 2 stores will create duplicated code

Reply:
I am not sure but I trust you on this one.